src/test/java/aniela/remanent/pdf/report/netto/PozycjeReportNettoFactory.java

```diff
@@ -0,0 +1,118 @@
+        });
+        return positions;
+    }
+
+    public List<PozycjaDoRaportuNetto> generateListOfPozycjaDoRaportuNettoContentMixed(int amountToCreate) {
+        List<PozycjaDoRaportuNetto> positions = new ArrayList<>(amountToCreate);
+        IntStream.range(1, amountToCreate).forEach(value -> {
+
+            if(value%2 == 0)
+                positions.add(createPozycjaDoRaportuNettoRandom(value));
+            else
+                positions.add(createPozycjaDoRaportuNetto(value));
+        });
+        return positions;
+    }
+
+
+
+
+    private PozycjaDoRaportuNetto createPozycjaDoRaportuNetto(int number) {
+        PozycjaDoRaportuNetto pozycjaDoRaportuNetto = new PozycjaDoRaportuNetto();
+        pozycjaDoRaportuNetto.setPozycjaWRaporcie(number);
+        pozycjaDoRaportuNetto.setNazwaTowaru(LONG_UNIT_NAME);
+        pozycjaDoRaportuNetto.setIlosc(ThreadLocalRandom.current().nextInt(1,20));
+        pozycjaDoRaportuNetto.setCenaNetto(3.11);
+        pozycjaDoRaportuNetto.setJednostka("szt");
+        pozycjaDoRaportuNetto.setSumaNetto(pozycjaDoRaportuNetto.getCenaNetto(), pozycjaDoRaportuNetto.getIlosc());
+        return pozycjaDoRaportuNetto;
+    }
+
+
+    private PozycjaDoRaportuNetto createPozycjaDoRaportuNettoRandom(int number) {
+        PozycjaDoRaportuNetto pozycjaDoRaportuNetto = new PozycjaDoRaportuNetto();
+        pozycjaDoRaportuNetto.setPozycjaWRaporcie(number);
+        pozycjaDoRaportuNetto.setNazwaTowaru(positionNames.getOrDefault(ThreadLocalRandom.current().nextInt(1,4),"Wartosc domyslna dla nazwy"));
+        pozycjaDoRaportuNetto.setIlosc(generatePseudoDouble());
+        pozycjaDoRaportuNetto.setCenaNetto(generatePseudoDouble());
+        pozycjaDoRaportuNetto.setJednostka(unitMapping.getOrDefault(ThreadLocalRandom.current().nextInt(1,4),"XXX"));
+        pozycjaDoRaportuNetto.setSumaNetto(pozycjaDoRaportuNetto.getCenaNetto(), pozycjaDoRaportuNetto.getIlosc());
+        return pozycjaDoRaportuNetto;
+    }
+
+
+    private Map<Integer, String> fillUnits() {
+        Map<Integer, String> units = new HashMap<>();
+        units.put(1, "szt");
+        units.put(2, "kpl");
+        units.put(3, "mb");
+        return units;
+    }
+
+    private Map<Integer, Double> fillNetto() {
+        Map<Integer, Double> mapping = new HashMap<>();
+        IntStream.range(1, 9).forEach(value -> {
+            double doubleRandom = BigDecimal.valueOf(ThreadLocalRandom.current().nextDouble(1, 245)).setScale(2,
+                RoundingMode.HALF_UP)
+                .doubleValue();
+            mapping.putIfAbsent(value, doubleRandom);
+        });
+        return mapping;
+    }
+
+    private Map<Integer, String> getPositions() {
+        Map<Integer, String> positions = new HashMap<>();
+        positions.put(1, LONG_UNIT_NAME);
+        positions.put(2, "cos dluzszego");
+        positions.put(3, "Znakow to tu bedzie jakies 29");
+        return positions;
+    }
+
+    public double generatePseudoDouble(){
```

Comment:
a to do czego ?

Reply:
a po to zeby wygenrowac liczby losowe netto. W tescie tworze 4 rodzaje raportow,
Odpal JUnit i zobacz ile raportow sie tworzy.

Reply:
W testach generuje 4 raporty zeby sprawdzic czy dobrze sumowanie dziala.
1 test case to za malo.